### PR TITLE
Prevent creation of SSL context during certificate updating

### PR DESCRIPTION
**Pull request by SKobziev**
Fixes #[2693](https://github.com/SmartDeviceLink/sdl_core/issues/2693)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes

### Summary
Added sync primitive into CryptoManagerImpl as member of class
and added lock into OnCertificateUpdated, CreateSSLContext
methods for preventing race conditions
Copy of https://github.com/smartdevicelink/sdl_core/pull/2704

In `OnCertificateUpdated` used functions which load the certificates and private keys into the SSL_CTX or SSL object, respectively. Namely, the `SSL_CTX_use_PrivateKey()` and `SSL_CTX_check_private_key()` functions loads the certificates and keys into the SSL_CTX object ctx. The information is passed to SSL objects ssl created from ctx with `SSL_new()` by copying, so that changes applied to ctx do not propagate to already existing SSL objects.  `SSL_new()`used in `CreateSSLContext`.  In order to avoid possible problems, it would be better if the streams of copying an object and data changes inside it did not intersect.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/security_manager/src/crypto_manager_impl.cc [hunk 1]
```diff
@@ -283,6 +283,9 @@ bool CryptoManagerImpl::Init() {
 
 bool CryptoManagerImpl::OnCertificateUpdated(const std::string& data) {
   LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(crypto_manager_lock_);
+  LOG4CXX_DEBUG(logger_,
+                "CryptoManager is locked. Start of sertificate update");
```
Comment (LuxoftAKutsan) [inline src/components/security_manager/src/crypto_manager_impl.cc]: @SKobziev This logs are redundant. Please Just swap Lock and autotrace

Comment (ydementieiev) [reply src/components/security_manager/src/crypto_manager_impl.cc]: @LuxoftAKutsan removed redundant logs [160fe94d](https://github.com/smartdevicelink/sdl_core/pull/2777/commits/160fe94dbd79df9521be91fea5f57551a896f607)

**Comment by ydementieiev** [comment]
@theresalech this PR passed internal Luxoft review and ready for Livio team review.

**Comment by AStasiuk** [comment]
@theresalech could you please tell me when it's planned to have PM review?

**Comment by iCollin** [comment]
@AStasiuk Hello, I just left a [comment on the issue](https://github.com/smartdevicelink/sdl_core/issues/2693#issuecomment-582544409) asking if the extra PTU was recently reproducible.

**Pull request merged**